qiskit/qpy/formats.py

```diff
@@ -94,14 +94,14 @@
         "num_parameters",
         "num_qargs",
         "num_cargs",
-        "has_condition",
+        "conditional_key",
         "condition_register_size",
         "condition_value",
         "num_ctrl_qubits",
         "ctrl_state",
     ],
 )
-CIRCUIT_INSTRUCTION_V2_PACK = "!HHHII?HqII"
+CIRCUIT_INSTRUCTION_V2_PACK = "!HHHIIBHqII"
```

Comment:
I assume you didn't version this pack string because `?`  and `B` are both a single byte and are interchangeable so nothing will break going from the old format to the new (except the parsing but it is a new format so that's fine).

Reply:
Ah, yep I see that called out in the docs too so yeah that makes sense.

Reply:
yeah, exactly - and the struct-pack conversion to `?` guarantees that the written value is exactly `0` or `1`, so there's precise ABI compatibility